pkg/model/components/kubelet.go

```diff
@@ -81,11 +81,17 @@ func (b *KubeletOptionsBuilder) BuildOptions(o interface{}) error {
 	}
 
 	if clusterSpec.Kubelet.ClusterDNS == "" {
-		ip, err := WellKnownServiceIP(clusterSpec, 10)
-		if err != nil {
-			return err
+		if clusterSpec.KubeDNS.NodeLocalDNS != nil {
+			if clusterSpec.KubeProxy.ProxyMode == "ipvs" && clusterSpec.KubeDNS.NodeLocalDNS.Enabled {
+				clusterSpec.Kubelet.ClusterDNS = "169.254.20.10"
```

Comment:
Instead of unconditionally setting the ClusterDNS IP here, it is better to add a check in the validation logic and get the user to set the correct address in the cluster spec.

Reply:
But how we would lead the user to set the correct address in the cluster spec? This is the address that it is recommended upstream. I believe is not an easy thing for the user to pick it up this address. They need to dig in the NodeLocal DNSCache documentation upstream. Which is your recommendation?

Reply:
The spec is validated against the rules defined in these files:
https://github.com/kubernetes/kops/tree/2e0720156ba193de4aca13fcb7c4776c04daedeb/pkg/apis/kops/validation

There should be a number of examples there that you can copy.
If any of these fail, the user will get information about what spec that needs to correct, to what, and why.